vpr/src/noc/turn_model_routing.cpp

```diff
@@ -233,3 +233,33 @@ TurnModelRouting::Direction TurnModelRouting::select_direction_other_than(const
     // if there was not any direction different from "other_than", return INVALID
     return TurnModelRouting::Direction::INVALID;
 }
+
+std::vector<std::pair<NocLinkId, NocLinkId>> TurnModelRouting::get_all_illegal_turns(const NocStorage& noc_model) const {
+    std::vector<std::pair<NocLinkId, NocLinkId>> illegal_turns;
+
+    for (const auto& noc_router : noc_model.get_noc_routers()) {
```

Comment:
Explain that you are getting all possible sets of 3 routers that could be reached in series (connected by links) and storing whether the resulting turn of the two links connecting them is legal given the a specified routing algorithm.